hawk/lib/hb_report.rb

```diff
@@ -0,0 +1,113 @@
+# Copyright (c) 2009-2015 Tim Serong <tserong@suse.com>
+# See COPYING for license.
+
+class HbReport
+  # Note: outfile, errfile are based off path passed to generate() -
+  # don't use them prior to a generate run, or you'll get the wrong path.
+  # Lastexit is global (this is freaky/dumb - everything should be based
+  # off path, and callers need to be updated to understand this - this
+  # will happen when we allow multiple hb_report runs, as
+  # hb_reports_controller is what cares about lastexit)
+  attr_reader :path
+  attr_reader :outfile
+  attr_reader :errfile
+  attr_reader :lastexit
+
+  def initialize(name = nil)
```

Comment:
Metrics/AbcSize: Assignment Branch Condition size for initialize is too high. [30.72/15]